### PR TITLE
revert union changes

### PR DESCRIPTION
**Pull request by ferozco**
Revert #22. Since it was originally just an experiment and it caused #44, I hope its ok. I am keen to open up the discussion on how implement unions, especially if we have some numbers around the performance of one approach vs. another

### CODE REVIEW

**Review by markelliot — APPROVED** [review]
👍

**Review by dansanduleac — APPROVED** [review]

File: conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanGenerator.java [hunk 1]
```diff
@@ -340,7 +301,9 @@ private static boolean isOptional(FieldSpec spec) {
 
         @Value.Derived
         default String getterName() {
-            return asGetterName(fieldName().get());
```
Comment (dansanduleac) [inline conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanGenerator.java]: With this refactor, asGetterName is now dead code

File: conjure-java-core/src/test/resources/example-types.yml [hunk 2]
```diff
@@ -160,3 +160,6 @@ types:
         union:
```
Comment (dansanduleac) [inline conjure-java-core/src/test/resources/example-types.yml:160]: do we need this `Union` type at all, given that there is also a `UnionTypeExample` type that does a similar thing but with more variants?

Comment (dansanduleac) [reply conjure-java-core/src/test/resources/example-types.yml:160]: Note: it's not used anywhere

**Review by dansanduleac — CHANGES_REQUESTED** [review]

**Review by dansanduleac — APPROVED** [review]

File: conjure-java-core/src/test/resources/example-types.yml (hunk 2 above)
Comment (dansanduleac) [reply conjure-java-core/src/test/resources/example-types.yml:160]: Left it there so we can have more cases covered by tests

**Pull request merged**